pkg/quarterdeck/db/models/users.go

```diff
@@ -819,6 +812,88 @@ func (u *User) SwitchOrganization(ctx context.Context, orgID any) (err error) {
 	return tx.Commit()
 }
 
+const (
+	deleteUserOrgSQL           = "DELETE FROM organization_users WHERE user_id=:userID AND organization_id=:orgID"
+	deleteUserInviteByEmailSQL = "DELETE FROM user_invitations WHERE email=:email AND organization_id=:orgID"
+)
+
+// RemoveOrganization removes the user from the specified organization. If this results
+// in the user having no organizations then the user is deleted. This also deletes all
+// invitations that have been sent to the user for the organization and all api keys
+// created by the user.
+func (u *User) RemoveOrganization(ctx context.Context, orgID any) (err error) {
+	var userOrg ulid.ULID
+	if userOrg, err = ulids.Parse(orgID); err != nil {
+		return err
+	}
+
+	if ulids.IsZero(userOrg) {
+		return ErrMissingOrgID
+	}
+
+	// Revoke all the keys the user created for this organization
+	if err = u.RevokeKeys(ctx, userOrg); err != nil {
+		return err
+	}
```

Comment:
I would like revocation to happen inside of the transaction; having in-memory work is going to be a no-go when we Raft-replicate Quarterdeck. Is the `RevokeKeys` function being used anywhere else?

Reply:
I created the function for this PR so it can definitely be changed to a transaction method without much disruption.